server/services/DeckService.js

```diff
@@ -37,13 +37,23 @@ class DeckService {
         return this.decks.find({ standaloneDeckId: { $exists: true } }, { sort: { lastUpdated: -1 } });
     }
 
-    create(deck) {
+    async create(deck) {
+        //if the eventId is set on a new deck, check if the user already has a deck with the same eventId
+        if(deck.eventId) {
+            //if a deck for the event already exists, do not create the new deck
+            if(this.userAlreadyHasDeckForEvent(deck.userName, deck.eventId)) {
+                return () => Promise.resolve();
+            }
+        }
+        
         let properties = {
             username: deck.username,
             name: deck.deckName,
             plotCards: deck.plotCards,
             bannerCards: deck.bannerCards,
             drawCards: deck.drawCards,
+            eventId: deck.eventId,
+            locked: deck.eventId ? true : false, // lock the deck from further changes if the eventId is set
```

Comment:
This shouldn't be a property of the deck itself, it should be computed when it's needed

Reply:
I see now that it's used in the client, so if you want to keep that, compute this property when fetching decks

Reply:
how would i go about computing it when fetching decks? do i need to add a handler to every find() method and add a property to the returned decks or is there some base functionality i can use that adds computed properties to fetched entities?